### PR TITLE
Add neighbor list unit test

### PR DESCRIPTION
**Pull request by streeve**
These tests are very similar to those in Cabana.

This also ensures that we don't test with more than one MPI rank (except for the future `Comm` test) and updates unit_test file license years.

One more towards #44

### CODE REVIEW

File: unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -46,4 +50,7 @@ macro(CabanaMD_add_tests)
   endforeach()
 endmacro()
 
-CabanaMD_add_tests(MPI NAMES Integrator)
+CabanaMD_add_tests(NAMES Integrator Neighbor)
```
Comment (sslattery) [inline unit_test/CMakeLists.txt:53]: Will you add the `MPI` case back here for tests that require multiple ranks?

File: unit_test/tstNeighbor.hpp [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+        }
+    };
+    Kokkos::parallel_for( exec_policy, fill_op );
+    Kokkos::fence();
+    return list;
+}
+
+//---------------------------------------------------------------------------//
+template <class ListType, class PositionSlice>
+void checkFullNeighborListPartialRange( const ListType &list,
```
Comment (sslattery) [inline unit_test/tstNeighbor.hpp:143]: I'm wondering if we can combine the full and partial test to reduce code here but I think this is OK for now as it is essentially what I implemented in Cabana

**Review by sslattery — APPROVED** [review]

File: unit_test/CMakeLists.txt (hunk 1 above)
Comment (streeve) [reply unit_test/CMakeLists.txt:53]: Yes, that should only be the Comm class

File: unit_test/tstNeighbor.hpp (hunk 2 above)
Comment (streeve) [reply unit_test/tstNeighbor.hpp:143]: Yeah, I should do that both here and in Cabana

**Pull request merged**